demos/speech_recognition_demo/python/utils/deep_speech_seq_pipeline.py

```diff
@@ -0,0 +1,98 @@
+            frame_window_size_seconds (float, in seconds)
+            frame_stride_seconds (float, in seconds)
+            mel_num (int)
+            mel_fmin (float, in Hz)
+            mel_fmax (float, in Hz)
+            num_mfcc_dct_coefs (int)
+            num_context_frames (int)
+            in_state_c, in_state_h, out_state_c, out_state_h, in_data, out_data (str), IR node names
+            log_probs (bool), True is input data contains base e log(probabilities), False if simply probabilities.
+        ie (IECore or None), IECore object to run NN inference with.  Default is to use ie_core_singleton module.
+            (default None)
+        device (str), inference device for IE, passed here to 1. set default device, and 2. check supported node types
+            in the model load; None = do not check (default 'CPU')
+        ie_extensions (list(tuple(str,str))), list of IE extensions to load, each extension is defined by a pair
+            (device, filename). Records with filename=None are ignored.  (default [])
+        online_decoding (bool), set to True to return partial decoded text after every input data piece (default False)
+        """
+        assert profile is not None, "profile argument must be provided"
+        self.p = deepcopy(profile)
+        self.mfcc_stage = AudioFeaturesSeqPipelineStage(profile)
+        self.rnn_stage = RnnSeqPipelineStage(model, model_bin=model_bin, profile=profile,
+                ie=ie, device=device, ie_extensions=ie_extensions)
+        self.ctc_stage = CtcDecoderSeqPipelineStage(lm=lm, profile=profile, beam_width=beam_width,
+                max_candidates=max_candidates, online=online_decoding)
+
+        super().__init__([
+            self.mfcc_stage,
+            self.rnn_stage,
+            self.ctc_stage,
+        ])
+
+    def activate_model(self, device):
+        self.rnn_stage.activate_model(device)
+
+    def recognize_audio(self, audio, sampling_rate, finish=True):
+        """
+        Run a segment of audio through ASR pipeline.
+        Use finish=True (default) to run recognition once for a whole utterance.
+        Use finish=False for online recognition, provinging data cut into segments and getting updated recognition
+        result after each segment. Set finish=True for the last segment to get the final result and reset pipeline state.
+        """
+        if audio is not None:
+            if abs(sampling_rate - self.p['model_sampling_rate']) > self.p['model_sampling_rate'] * 0.1  or  (audio.shape + (1,))[1] != 1:
+                raise ValueError("Input audio file should be {} kHz mono".format(self.p['model_sampling_rate']/1e3))
+        return self.process_data(audio, finish=finish)
+
+    def extract_mfcc(self, audio, sampling_rate, finish=True):
+        # Audio feature extraction
+        if audio is not None:
+            if abs(sampling_rate - self.p['model_sampling_rate']) > self.p['model_sampling_rate'] * 0.1  or  (audio.shape + (1,))[1] != 1:
```

Comment:
Why do you allow 10% percent deviation in sample rate?

Reply:
1. Because, for example, 1 Hz difference obviously doesn't affect anything, and I've seen such files in the past, which were 1 Hz off some standard frequency.  2. I think up to 10% change in sampling rate (which is equivalent to 10% = ~1 tone change in voice pitch and pace) is probably within the normal variation of speakers, and should be OK.